Replacing non-streaming audio with live audio

carlosapaduarte opened the pull request:
Updating '`audio` element content has text alternative' (#e7aa44) so that it no longer uses the non-streaming definition which was found to be inaccurate

I'm looking for comments on:
- The appropriateness of the expectation in the new atomic rule that checks if the audio content is live
- Suggestions on resources to use for the passed example

After getting this change approved I'll open a new PR changing the other rules that also use the non-streaming definition

Need for Final Call:
This will require a 2 weeks Final Call

---

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

carlosapaduarte committed:
replacing non-streaming audio with live audio - initial commit

carlosapaduarte committed:
Update spelling-ignore.yml

ajanec01 commented:
I must've missed that part, what's inaccurate with "non-streaming"?

carlosapaduarte commented:
This: https://github.com/act-rules/act-rules.github.io/issues/1446#issue-701040347

carlosapaduarte commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -0,0 +1,91 @@
+
+## Background
+
+- [Understanding SC 1.2.1:Audio-only and Video-only (Prerecorded)](https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded)
```
I think I'll address this one after sorting out this rule.

carlosapaduarte commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -0,0 +1,91 @@
+
+## Expectation 
+
+The target element's [media resource](https://html.spec.whatwg.org/multipage/media.html#media-resource) is being broadcasted in real time without being first recorded and stored on the origin server.
```
Changing the expectation.

carlosapaduarte commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -0,0 +1,91 @@
+This rule applies to every `audio` element that is:
+
+- playing; or,
+- has a "play button" that is [visible][] and [included in the accessibility tree][].
```
It sounds reasonable to have an atomic for all media, but I'll also address this after sorting this rule first.

carlosapaduarte committed:
Create play-button.md

carlosapaduarte committed:
update expectation

carlosapaduarte committed:
comments about the passed example

Jym77 commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -19,13 +19,13 @@ acknowledgments:
 
-The target element's [media resource](https://html.spec.whatwg.org/multipage/media.html#media-resource) is being broadcasted in real time without being first recorded and stored on the origin server.
+For each test target one of the following is true:
+- it is not possible to [seek][] the end of the target element's [media resource][]; or 
```
I like that 👍 
OTOH, I'm not sure it's possible to "not be possible to seek". The algorithm seems to always return (possibly after doing nothing). Would that rather be part of the second bullet (and so the first can be removed)?

carlosapaduarte replied:
I believe you are correct. I've updated the expectation accordingly.

carlosapaduarte committed:
Add passed example with live audio from microphone on device

carlosapaduarte committed:
Update expectation

Addresses https://github.com/act-rules/act-rules.github.io/pull/1661/files#r674724648

carlosapaduarte commented:
Updated rule, now with a working example.

Jym77 commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -0,0 +1,123 @@
+
+This `audio` element plays a live media resource by streaming the audio from the device's microphone.
+
+This example requires a browser supporting the experimental MediaStreamTrack API. The browser must be authorized to access a microphone. You can successfully test this example on Google Chrome with the flag --enable-blink-features=WebCodecs,MediaStreamInsertableStreams
```
I think we should (also) put that text inside the HTML page of the example itself.
That way, (manual) testers that navigate to the example's page (without necessarily going through the ACT rules page) would be able to see that text and act accordingly.

Jym77 replied:
This now seems related to #1679 since "give access to your mic" sounds a bit like a test instruction 🤔

carlosapaduarte replied:
* Moved the note to the example so that manual testers are aware of the instructions
* Added a test instructions section after the description of the example

Jym77 commented on _rules/audio-is-live-213x3x.md:
```diff
@@ -0,0 +1,123 @@
+
+#### Passed Example 1
+
+This `audio` element plays a live media resource by streaming the audio from the device's microphone.
```
Also, have you tested that by "seeking" the end, to make sure this matches the expectation?
Wondering if the "seek" bit should be used also in examples descriptions 🤔

carlosapaduarte replied:
I just found that MediaStream is not seekable! https://w3c.github.io/mediacapture-main/#mediastreams-in-media-elements 
I will add this to the expectation and the example's description.

carlosapaduarte replied:
And I suppose this lands us again exactly where we were before... needing to find a live stream that we can use in the example 😞

carlosapaduarte committed:
Merge branch 'develop' into live-media-resource

carlosapaduarte committed:
update description and content of PE1 to include instructions

carlosapaduarte committed:
Update spelling-ignore.yml

carlosapaduarte committed:
update expectation and PE1 to handle MediaStream not being seekable

carlosapaduarte committed:
Update spelling-ignore.yml

carlosapaduarte committed:
Update spelling-ignore.yml

carlosapaduarte committed:
Fix conflicts

carlosapaduarte committed:
Update audio-text-alternative-e7aa44.md

daniel-montalvo commented on _rules/audio-is-live-213x3x.md:
````diff
@@ -0,0 +1,140 @@
+	</script>
+</html>
+```
+
````
I am assuming we are expanding these passed examples with real streaming audios/videos of broadcasting orgs, right?

carlosapaduarte replied:
That was the main idea. 

As was also mentioned in TPAC, we might want to explore the possibility of W3C hosting a streaming server where we could serve the examples (we would just loop one of the videos we already use).
